File: Part 2 - Regression/Section 4 - Simple Linear Regression/problems/autoinsurence/solution.R
#' 
#' Created on Sun Mar 11 23:03:35 2018
#' 
#' @author: mural.kummitha
#' 
# ======> Auto Insurance in Sweden simple linear regression

# ====> read dataset
library(caTools)
library(ggplot2)
setwd('~/work/mlatoz/Part 2 - Regression/Section 4 - Simple Linear Regression/problems/autoinsurence')
dataset = read.csv('AutoInsurSweden.csv')


# ====> separate data into training set and test set
set.seed(123)
split = sample.split(dataset$TotalPayment, SplitRatio = 0.8)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# ====> predict the simple linear regression
regressor = lm(formula = TotalPayment ~ NoOfClaims, data = training_set)
payment_predict = predict(regressor, test_set)
payment_training_set_predict = predict(regressor, training_set)

# ====> plot training set regression graph
training_set_plot = ggplot() + 
                      geom_point(aes(x=training_set$NoOfClaims, y=training_set$TotalPayment), color='red') +
                      geom_line(aes(x=training_set$NoOfClaims, y=payment_training_set_predict), color='blue') +
                      ggtitle('No.of claims vs Payment (Training set) ') +
                      xlab('No.of Claims') + 
                      ylab('Payment')
ggsave(filename = 'r_training_set.png', plot = training_set_plot)


# ====> plot test set regression graph
test_set_plot = ggplot() +
                  geom_point(aes(x=test_set$NoOfClaims, y=test_set$TotalPayment), color='red') +
                  geom_line(aes(x=training_set$NoOfClaims, y=payment_training_set_predict), color='blue') +
                  ggtitle('No.of Claims vs Payment (Test set) ') +
                  xlab('No.of Claims') + 
                  ylab('Payment')

ggsave(filename = 'r_test_set.png', plot=test_set_plot)

File: Part 2 - Regression/Section 4 - Simple Linear Regression/problems/autoinsurence/solution.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Sun Mar 11 23:03:35 2018

@author: mural.kummitha
"""
# ======> Auto Insurance in Sweden simple linear regression

from sklearn.cross_validation import train_test_split
from sklearn.linear_model import LinearRegression
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

# =====> read dataset
dataset = pd.read_csv('AutoInsurSweden.csv')

# ====> split data set into training set and test set
no_of_claims = dataset.iloc[:, :-1].values
payment = dataset.iloc[:, 1].values
no_of_claims_train, no_of_claims_test, payment_train, payment_test = train_test_split(no_of_claims, payment, test_size=0.2)

# =====> fitting simple linear to the training set
regressor = LinearRegression()
regressor.fit(no_of_claims_train, payment_train)
payment_predict = regressor.predict(no_of_claims_test)


# =====> Plot training set regression graph
plt.scatter(no_of_claims_train, payment_train, color='red')
plt.plot(no_of_claims_train, regressor.predict(no_of_claims_train), color='blue')
plt.xlabel('No of Claims')
plt.ylabel('Total Payment')
plt.title('No.of Claims vs Total Payment (Training set)')
plt.savefig('p_train_set.png')
plt.close()


# ====> Plot test set regression grapth
plt.scatter(no_of_claims_test, payment_test, color='red')
plt.plot(no_of_claims_train, regressor.predict(no_of_claims_train), color='blue')
plt.title('No.of Claims vs Total Payment (Test set)')
plt.xlabel('No of Claims')
plt.ylabel('Total Payment')
plt.savefig('p_test_set.png')
plt.close()